lib/SILOptimizer/Analysis/MemoryBehavior.cpp

```diff
@@ -566,6 +566,27 @@ static SILValue getBeginScopeInst(SILValue V) {
   if (BorrowedValue borrowedObj = getSingleBorrowIntroducingValue(object)) {
     return borrowedObj.value;
   }
+
+  if (!object->getFunction()->hasOwnership()) {
+    // In non-OSSA, do a quick check if the object is a guaranteed function
+    // argument.
+    // Note that in OSSA, getSingleBorrowIntroducingValue will detect a
+    // guaranteed argument.
+    for (;;) {
```

Comment:
This looks like a partial reimplementation of `findOwnershipReferenceAggregate`

Reply:
👍 Using `findOwnershipReferenceAggregate` now